sklearn/metrics/tests/test_pairwise.py

```diff
@@ -407,6 +407,28 @@ def test_euclidean_distances():
     assert_greater(np.max(np.abs(wrong_D - D1)), .01)
 
 
+def test_cosine_distances():
+    # Check the pairwise cosine_distances distances computation
+    rng = np.random.RandomState(1337)
+    x = np.abs(rng.rand(910))
+    XA = np.vstack([x, x])
+    D = cosine_distances(XA)
+    assert_array_almost_equal(D, [[0., 0.], [0., 0.]])
+    # check that all elements are in [0, 2]
+    assert_true(np.all(D >= 0.))
```

Comment:
>  I guess the >=0 checks if it's actually >= and not only "almost"

Absolutely. If teh value is -1e-16 than it will pass `assert_array_almost_equal`, but the second check on `>=0` will make it fail.

OK. I will add another test with random matrix 1000 x 5000 .